sql/hive/src/test/scala/org/apache/spark/sql/sources/HadoopFsRelationTest.scala

```diff
@@ -138,7 +137,8 @@ abstract class HadoopFsRelationTest extends QueryTest with SQLTestUtils with Tes
           logInfo(s"Testing $dataType data type$extraMessage")
 
           val extraOptions = Map[String, String](
-            "parquet.enable.dictionary" -> parquetDictionaryEncodingEnabled.toString
+            "parquet.enable.dictionary" -> parquetDictionaryEncodingEnabled.toString,
+            "timestampFormat" -> "yyyy-MM-dd'T'HH:mm:ss.SSSXXXXX"
```

Comment:
with the new parser and the default timestamp format, spark can't write and read back timestamp data before 1582?

what's the consequence if we make this the default format?

Reply:
Similar question was raised at https://github.com/apache/spark/pull/23417#discussion_r244549254. Looks this is going to be investigated soon separately.

It's going to at least introduce some behaviour changes:

```scala
scala> val fomatter = new org.apache.spark.sql.catalyst.util.Iso8601TimestampFormatter("yyyy-MM-dd'T'HH:mm:ss.SSSXXX", java.util.TimeZone.getDefault(), java.util.Locale.US)
fomatter: org.apache.spark.sql.catalyst.util.Iso8601TimestampFormatter = org.apache.spark.sql.catalyst.util.Iso8601TimestampFormatter@2df019b8

scala> fomatter.format(fomatter.parse("0015-03-10T08:53:43.591+07:30"))
res0: String = 0015-03-10T08:19:08.591+06:55

scala> val fomatter = new org.apache.spark.sql.catalyst.util.Iso8601TimestampFormatter("yyyy-MM-dd'T'HH:mm:ss.SSSXXXXX", java.util.TimeZone.getDefault(), java.util.Locale.US)
fomatter: org.apache.spark.sql.catalyst.util.Iso8601TimestampFormatter = org.apache.spark.sql.catalyst.util.Iso8601TimestampFormatter@763ff91

scala> fomatter.format(fomatter.parse("0015-03-10T08:53:43.591+07:30"))
res1: String = 0015-03-10T08:19:08.591+06:55:25
```

Reply:
Yeah this is a complicated issue. But I think one takeaway is that old dates weren't actually being written correctly before. Without the seconds on the timezone, the first date can't be read back correctly as that info is lost.